backend/packages/Upgrade/src/api/controllers/validators/LogValidator.ts

```diff
@@ -1,14 +1,57 @@
-import { IsNotEmpty, IsDefined, IsString, IsJSON } from 'class-validator';
-import { ILogInput } from 'upgrade_types';
+import { Type } from 'class-transformer';
+import { IsNotEmpty, IsDefined, IsString, ValidateNested, IsArray, IsOptional, IsObject } from 'class-validator';
+
+class ILogGroupMetrics {
+  @IsString()
+  @IsNotEmpty()
+  groupClass: string;
+
+  @IsString()
+  @IsNotEmpty()
+  groupKey: string;
+
+  @IsString()
+  @IsNotEmpty()
+  groupUniquifier: string;
+
+  @IsObject()
+  attributes: any;
+}
+
+class ILogMetrics {
+  @IsOptional()
+  @IsObject()
+  attributes: any;
```

Comment:
If this is optional can you change it to attributes?:any. And can we replace the any with Record<string, number|string>. I don't think we need any other data type in the value other than number and string.

Reply:
Code updated